### DISCUSSION TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'yarn'

### DISCUSSION THREAD

**Discussion by eranirudhkumar** [dask/dask#7316] [Q&A]
Hi, I am trying with Dask YarnCluster but getting an error- "No such file or directory "yarn":

OS: Ubuntu 20.04
conda: 4.9.2
Python: 3.8.8
Name                    Version                   Build             Channel
dask-core                 2021.2.0           pyhd8ed1ab_0    conda-forge
dask-yarn                 0.8.1            py38h32f6830_0    conda-forge

~$ cat ~/.config/dask/yarn.yaml
      ...................

~$ ipython
Python 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.21.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from dask_yarn import YarnCluster

In [2]: cluster = YarnCluster(environment='Dask.tar.gz', worker_vcores=1, worker_memory='1GB')
........................................................................................................................................................
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-2-b5edf354063f> in <module>
----> 1 cluster = YarnCluster(environment='Dask.tar.gz', worker_vcores=1, worker_memory='1GB')

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in __init__(self, environment, n_workers, worker_vcores, worker_memory, worker_restarts, worker_env, scheduler_vcores, scheduler_memory, deploy_mode, name, queue, tags, user, host, port, dashboard_address, skein_client, asynchronous, loop)
    381             user=user,
    382         )
--> 383         self._init_common(
    384             spec=spec,
    385             host=host,

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in _init_common(self, spec, application_client, host, port, dashboard_address, asynchronous, loop, skein_client)
    531
    532         if not self.asynchronous:
--> 533             self._sync(self._start_internal())
    534
    535     def _start_cluster(self):

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in _sync(self, task)
    699         future = asyncio.run_coroutine_threadsafe(task, self.loop.asyncio_loop)
    700         try:
--> 701             return future.result()
    702         except BaseException:
    703             future.cancel()

~/anaconda3/envs/Dask/lib/python3.8/concurrent/futures/_base.py in result(self, timeout)
    437                 raise CancelledError()
    438             elif self._state == FINISHED:
--> 439                 return self.__get_result()
    440             else:
    441                 raise TimeoutError()

~/anaconda3/envs/Dask/lib/python3.8/concurrent/futures/_base.py in __get_result(self)
    386     def __get_result(self):
    387         if self._exception:
--> 388             raise self._exception
    389         else:
    390             return self._result

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in _start_internal(self)
    590             self._start_task = asyncio.ensure_future(self._start_async())
    591         try:
--> 592             await self._start_task
    593         except BaseException:
    594             # On exception, cleanup

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in _start_async(self)
    605             else:
    606                 self._scheduler = None
--> 607             await self.loop.run_in_executor(None, self._start_cluster)
    608         else:
    609             # Connect to an existing cluster

~/anaconda3/envs/Dask/lib/python3.8/concurrent/futures/thread.py in run(self)
     55
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in _start_cluster(self)
    536         """Start the cluster and initialize state"""
    537
--> 538         skein_client = _get_skein_client(self._skein_client)
    539
    540         if "dask.scheduler" not in self.spec.services:

~/anaconda3/envs/Dask/lib/python3.8/site-packages/dask_yarn/core.py in _get_skein_client(skein_client, security)
    112         with warnings.catch_warnings():
    113             warnings.simplefilter("ignore")
--> 114             return skein.Client(security=security)
    115     return skein_client
    116

~/anaconda3/envs/Dask/lib/python3.8/site-packages/skein/core.py in __init__(self, address, security, keytab, principal, log, log_level, java_options)
    336
    337         if address is None:
--> 338             address, proc = _start_driver(security=security,
    339                                           keytab=keytab,
    340                                           principal=principal,

~/anaconda3/envs/Dask/lib/python3.8/site-packages/skein/core.py in _start_driver(security, set_global, keytab, principal, log, log_level, java_options)
    215     env['SKEIN_KEY'] = security._get_bytes('key')
    216     # Update the classpath in the environment
--> 217     classpath = (subprocess.check_output(['yarn', 'classpath', '--glob'])
    218                            .decode('utf-8'))
    219     env['CLASSPATH'] = '%s:%s' % (_SKEIN_JAR, classpath)

~/anaconda3/envs/Dask/lib/python3.8/subprocess.py in check_output(timeout, *popenargs, **kwargs)
    413         kwargs['input'] = empty
    414
--> 415     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    416                **kwargs).stdout
    417

~/anaconda3/envs/Dask/lib/python3.8/subprocess.py in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    491         kwargs['stderr'] = PIPE
    492
--> 493     with Popen(*popenargs, **kwargs) as process:
    494         try:
    495             stdout, stderr = process.communicate(input, timeout=timeout)

~/anaconda3/envs/Dask/lib/python3.8/subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors, text)
    856                             encoding=encoding, errors=errors)
    857
--> 858             self._execute_child(args, executable, preexec_fn, close_fds,
    859                                 pass_fds, cwd, env,
    860                                 startupinfo, creationflags, shell,

~/anaconda3/envs/Dask/lib/python3.8/subprocess.py in _execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, start_new_session)
   1704                     if errno_num != 0:
   1705                         err_msg = os.strerror(errno_num)
-> 1706                     raise child_exception_type(errno_num, err_msg, err_filename)
   1707                 raise child_exception_type(err_msg)
   1708

**FileNotFoundError: [Errno 2] No such file or directory: 'yarn'**

**Comment by jsignell** [comment]
This looks like an issue with dask-yarn. I suggest you open a bug report in that repository https://github.com/dask/dask-yarn/issues
